catalyst-gateway/bin/src/service/api/cardano/rbac/registrations_get/role_data.rs

```diff
@@ -0,0 +1,104 @@
+//! A role data.
+
+use std::collections::HashMap;
+
+use anyhow::Context;
+use cardano_blockchain_types::Slot;
+use poem_openapi::{types::Example, Object};
+use rbac_registration::{
+    cardano::cip509::{PointData, RoleData},
+    registration::cardano::RegistrationChain,
+};
+
+use crate::{
+    service::api::cardano::rbac::registrations_get::{
+        extended_data::ExtendedData, key_data::KeyData, key_data_list::KeyDataList,
+        payment_data::PaymentData, payment_data_list::PaymentDataList,
+    },
+    settings::Settings,
+};
+
+/// A RBAC registration role data.
+#[derive(Object, Debug, Clone)]
+#[oai(example)]
+pub struct RbacRoleData {
+    /// A list of role signing keys.
+    #[oai(skip_serializing_if_is_empty)]
+    signing_keys: KeyDataList,
+    /// A list of role encryption keys.
+    #[oai(skip_serializing_if_is_empty)]
+    encryption_keys: KeyDataList,
+    /// A list of role payment addresses.
+    #[oai(skip_serializing_if_is_empty)]
+    payment_addresses: PaymentDataList,
+    /// A map of the extended data.
+    ///
+    /// Unlike other fields, we don't track history for this data.
```

Comment:
Just curious why we are not tracking the history of the extended data?

Reply:
Unless it was a misunderstanding from my side, it was an input from Steven.

@stevenj Please confirm. 🙃 If it was indeed a misunderstanding, I will update the PR.